node/peers.js

```diff
@@ -41,19 +41,24 @@ function TChannelPeers(channel, options) {
     self.options = options || {};
     self.initialPeerState = self.options.initialPeerState;
     self._map = Object.create(null);
-    self.selfPeer = TChannelSelfPeer(self.channel);
+    self.selfPeer = null;
 }
 
 inherits(TChannelPeers, EventEmitter);
 
 TChannelPeers.prototype.close = function close(callback) {
     var self = this;
-    var peers = [self.selfPeer].concat(self.values());
-    var counter = peers.length;
+
+    var peers = self.values();
+    if (self.selfPeer) {
+        peers.push(self.selfPeer);
+    }
+    var counter = peers.length + 1;
     peers.forEach(function eachPeer(peer) {
         peer.close(onClose);
     });
     self.clear();
+    onClose();
 
     function onClose() {
```

Comment:
I set counter to be length + 1 and we call `onClose` for each peer; I call an extra `onClose` at the end.

This means that Peers will close if there are zero peers;

The alternative is `if (peers.length === 0) { cb(); }`